### DISCUSSION TITLE
PyRFC and Classes

### DISCUSSION THREAD

**Discussion by the-master-student** [SAP/PyRFC#264] [Q&A] [answered]
Hi everyone,

I am trying to extract the os on which the SAP GUI is running. As explained in this thread (https://answers.sap.com/questions/9623610/find-the-operating-system-of-a-user.html) I can do this using the global class CL_GUI_FRONTEND_SERVICES to get information about the presentation layer. I lack ABAP knowledge and was wondering if there is a way to call methods and not function modules using pyrfc. 

Thank you so much in advance.

**Comment by bsrdjan** [comment, marked as answer]
RFC connectors cannot call class methods. Custom ABAP RFC can be created, internally calling class method of interest.

The function module might be also already available. ABAP developer can check if that is the case or you can check SAP ABAP Community and post the question there, if there is a FM or RFM exposing CL_GUI_FRONTEND_SERVICES.

**Reply by the-master-student** [reply]
Thank you so much :)
